### PR TITLE
[plugin][pulsar] Handle the SendCallback method sign changed.

### PR DESCRIPTION
**Pull request by dao-jun**
In https://github.com/apache/pulsar/pull/22940, Pulsar will change the `org.apache.pulsar.client.impl.SendCallback#sendComplete` method sign from

```java
public void sendComplete(Exception e)
```
to

```java
public void sendComplete(Throwable t, OpSendMsgStats stats)
```

This PR is to address it.

- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking-java/blob/main/CHANGES.md).

### CODE REVIEW

**Comment by wu-sheng** [comment]
Please fix CI and we need to add the new versions to the test scenarios to verify.

Also, you need to make CI passed.

File: CHANGES.md [hunk 1]
```diff
@@ -6,6 +6,7 @@ Release Notes.
 ------------------
 
 * Upgrade nats plugin to support 2.16.5
+* [pulsar] Handle the SendCallback class method signature change.
```
Comment (wu-sheng) [inline CHANGES.md:9]:
```suggestion
* Support newer Pulsar versions to adopt SendCallback class method signature change.
```

I can't see which versions will include this. The PR you refer to doesn't have the version information.

Comment (dao-jun) [reply CHANGES.md:9]: The code implementation PR: https://github.com/apache/pulsar/pull/23196.
It will available in the next release.

Comment (wu-sheng) [reply CHANGES.md:9]: We can't accept this PR until it is released and verified through CI tests.

We have to hold this until then.

**Comment by wu-sheng** [comment]
Could you fix the CI first? Then we need to wait for Pulsar's next release.

**Comment by wu-sheng** [comment]
No update for a month. Close for now.

**Pull request closed without merging**